pkg/gameserversets/gameserversets_test.go

```diff
@@ -35,13 +35,19 @@ func TestSortGameServersByLeastFullNodes(t *testing.T) {
 	nc := map[string]gameservers.NodeCount{
 		"n1": {Ready: 1, Allocated: 0},
 		"n2": {Ready: 0, Allocated: 2},
+		"n3": {Ready: 2, Allocated: 0},
+		"n4": {Ready: 2, Allocated: 0},
 	}
 
 	list := []*agonesv1.GameServer{
 		{ObjectMeta: metav1.ObjectMeta{Name: "g1"}, Status: agonesv1.GameServerStatus{NodeName: "n2", State: agonesv1.GameServerStateReady}},
 		{ObjectMeta: metav1.ObjectMeta{Name: "g2"}, Status: agonesv1.GameServerStatus{NodeName: "", State: agonesv1.GameServerStateReady}},
 		{ObjectMeta: metav1.ObjectMeta{Name: "g3"}, Status: agonesv1.GameServerStatus{NodeName: "n1", State: agonesv1.GameServerStateReady}},
 		{ObjectMeta: metav1.ObjectMeta{Name: "g4"}, Status: agonesv1.GameServerStatus{NodeName: "n2", State: agonesv1.GameServerStateCreating}},
+		{ObjectMeta: metav1.ObjectMeta{Name: "g5"}, Status: agonesv1.GameServerStatus{NodeName: "n3", State: agonesv1.GameServerStateReady}},
+		{ObjectMeta: metav1.ObjectMeta{Name: "g6"}, Status: agonesv1.GameServerStatus{NodeName: "n4", State: agonesv1.GameServerStateReady}},
+		{ObjectMeta: metav1.ObjectMeta{Name: "g7"}, Status: agonesv1.GameServerStatus{NodeName: "n3", State: agonesv1.GameServerStateReady}},
+		{ObjectMeta: metav1.ObjectMeta{Name: "g8"}, Status: agonesv1.GameServerStatus{NodeName: "n4", State: agonesv1.GameServerStateReady}},
```

Comment:
Is the relative order of g5 and g7, g6 and g8 guaranteed, given `sort.Slice` is not stable?

Reply:
Run the test a few hundred times and see? 😁

Reply:
Surely sort.Slice is not stable!
I have run this several hundred times and the results are stable, however there are no guarantees so I will change the assertion. Thanks!